charts/provisioner/templates/cronjob.yaml

```diff
@@ -0,0 +1,319 @@
+            checksum/manifests-cm: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum  | trunc 63 }}
+            checksum/config-secrets: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum | trunc 63 }}
+          {{- if .Values.podAnnotations }}
+          annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.podAnnotations "context" $) | nindent 12 }}
+          {{- end }}
+        spec:
+          {{- include "provisioner.imagePullSecrets" . | nindent 6 }}
+          {{- if .Values.schedulerName }}
+          schedulerName: {{ .Values.schedulerName }}
+          {{- end }}
+          restartPolicy: {{ .Values.restartPolicy }}
+          {{- if .Values.priorityClassName }}
+          priorityClassName: {{ .Values.priorityClassName }}
+          {{- end }}
+          {{- if .Values.topologySpreadConstraints }}
+          topologySpreadConstraints: {{- include "common.tplvalues.render" (dict "value" .Values.topologySpreadConstraints "context" $) | nindent 12 }}
+          {{- end }}
+          terminationGracePeriodSeconds: 0
+          {{- if .Values.topologySpreadConstraints }}
+          topologySpreadConstraints: {{- include "common.tplvalues.render" (dict "value" .Values.topologySpreadConstraints "context" .) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.podSecurityContext }}
+          securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.podSecurityContext "context" $) | nindent 12 }}
+          {{- end }}
+          serviceAccountName: {{ template "provisioner.serviceAccountName" . }}
+          initContainers:
+          {{- if .Values.console.enabled }}
+            - name: wait-for-console
+              image: {{ include "common.images.image" (dict "imageRoot" .Values.waitFor.image "global" .Values.global) }}
+              imagePullPolicy: {{ .Values.waitFor.image.pullPolicy | quote }}
+              command:
+               - "/bin/sh"
+               - "/wait-for.sh"
+              args:
+                - "{{ .Values.console.url }}/api/v1/health/ready"
+              env:
+                {{- if .Values.console.url}}
+                - name: CDK_BASE_URL
+                  value: {{ .Values.console.url | quote }}
+                {{- end }}
+              envFrom:
+                - secretRef:
+                    name: {{ include "provisioner.config.secrets" . }}
+                {{- if .Values.console.extraEnvVarsCM }}
+                - configMapRef:
+                    name: {{ include "common.tplvalues.render" (dict "value" .Values.console.extraEnvVarsCM "context" $) }}
+                {{- end }}
+                {{- if .Values.console.extraEnvVarsSecret }}
+                - secretRef:
+                    name: {{ include "common.tplvalues.render" (dict "value" .Values.console.extraEnvVarsSecret "context" $) }}
+                {{- end }}
```

Comment:
Why do the init-containers have the `secrets` and `configmaps` mounted in?
Does the healtcheck endpoint need authentication?

Reply:
No it's to resolve Console or Gateway URL env var that can be set directly on values or through configmap/secrets